src/main/data_structure/string_exercise.py

```diff
@@ -0,0 +1,71 @@
+from src.main.logger.py_logger import PyLogger
+
+logger = PyLogger.get_configured_logger()
+
+
+# Count the number of characters (character appearance) in a string
+def char_appearance(string):
+    char_dict = {}
+    for n in string:
+        keys = char_dict.keys()
+        if n in keys:
+            char_dict[n] += 1
+        else:
+            char_dict[n] = 1
+    return char_dict
+
+
+#  Add 'ing' at the end of a given string (length should be at least 3).
+#  If the given string already ends with 'ing' then add 'ly' instead.
+#  If the string length of the given string is less than 3, leave it unchanged
+def add_string(string):
+    length = len(string)
+    if length > 2:
+        if string[-3:] == 'ing':
+            string += 'ly'
+        else:
+            string += 'ing'
+    return string
+
+
+# Find the first appearance of the substring 'not' and 'poor' from a given string,
+# if 'not' follows the 'poor', replace the whole 'not'...'poor' substring with 'good'.
+# Return the resulting string
+def string_replace(string):
```

Comment:
change the function name. function name always should be verb

Reply:
ok dada